From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Icarus Sparry <icarus.w.sparry@intel.com>
Date: Tue, 25 Apr 2017 16:42:38 -0700
Subject: [PATCH] Convert to stateless configuration

At runtime if CONFFILE (/etc/dnsmasq.conf) doesn't exist then try
/usr/share/defaults/dnsmasq/dnsmasq.conf instead.

Signed-off-by: Icarus Sparry <icarus.w.sparry@intel.com>
---
 src/config.h | 8 ++++++++
 src/option.c | 4 +++-
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/src/config.h b/src/config.h
index 9f69ce3..12c2815 100644
--- a/src/config.h
+++ b/src/config.h
@@ -224,6 +224,14 @@ RESOLVFILE
 #   endif
 #endif
 
+#ifndef DEFAULT_CONFFILE
+#   if defined(__FreeBSD__)
+#      define DEFAULT_CONFFILE "/usr/local/share/defaults/dnsmasq/dnsmasq.conf"
+#   else
+#      define DEFAULT_CONFFILE "/usr/share/defaults/dnsmasq/dnsmasq.conf"
+#   endif
+#endif
+
 #ifndef RUNFILE
 #   if defined(__ANDROID__)
 #      define RUNFILE "/data/dnsmasq.pid"
diff --git a/src/option.c b/src/option.c
index 0a72406..9eddbea 100644
--- a/src/option.c
+++ b/src/option.c
@@ -5143,8 +5143,10 @@ void read_opts(int argc, char **argv, char *compile_opts)
       one_file(conffile, 0);
       free(conffile);
     }
-  else
+  else if (access(CONFFILE, F_OK) != -1)
     one_file(CONFFILE, LOPT_CONF_OPT);
+  else
+    one_file(DEFAULT_CONFFILE, LOPT_CONF_OPT);
 
   /* port might not be known when the address is parsed - fill in here */
   if (daemon->servers)
